crm_rma_advance_warranty/model/crm_claim_rma.py

```diff
@@ -0,0 +1,134 @@
+# -*- encoding: utf-8 -*-
+# ##############################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
+#    Copyright (C) OpenERP Venezuela (<http://www.vauxoo.com>).
+#    All Rights Reserved
+# ############ Credits ########################################################
+#    Coded by: Yanina Aular <yani@vauxoo.com>
+#    Planified by: Yanina Aular <yani@vauxoo.com>
+#    Audited by: Moises Lopez <moylop260@vauxoo.com>
+#                Nhomar Hernandez <nhomar@vauxoo.com>
+# ##############################################################################
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+
+from openerp import models, api
+from openerp.exceptions import except_orm
+from openerp.tools.translate import _
+from datetime import datetime
+# from dateutil.relativedelta import relativedelta
+from openerp.tools import (DEFAULT_SERVER_DATE_FORMAT,
+                           DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+class claim_line(models.Model):
+
+    _inherit = 'claim.line'
+
+    # Method to calculate warranty limit
+    @api.one
+    @api.model
+    def set_warranty_limit(self):
```

Comment:
If you put @api.model, by definition you can't put @api.one, because there's no information of the affected record in the arguments.